tensorflow_recommenders_addons/dynamic_embedding/python/kernel_tests/warm_start_util_test.py

```diff
@@ -0,0 +1,183 @@
+import os
+import shutil
+
+from tensorflow_recommenders_addons import dynamic_embedding as de
+
+try:
+  from tensorflow.python.keras.initializers import initializers_v2 as kinit2
+except ImportError:
+  kinit2 = None
+  pass  # for compatible with TF < 2.3.x
+
+from tensorflow.core.protobuf import cluster_pb2
+from tensorflow.core.protobuf import config_pb2
+from tensorflow.python.eager import context
+from tensorflow.python.framework import constant_op
+from tensorflow.python.framework import dtypes
+from tensorflow.python.framework import ops
+from tensorflow.python.framework import sparse_tensor
+from tensorflow.python.framework import tensor_shape
+from tensorflow.python.framework import test_util
+from tensorflow.python.keras import initializers as keras_init_ops
+from tensorflow.python.ops import array_ops
+from tensorflow.python.ops import embedding_ops
+from tensorflow.python.ops import gen_array_ops
+from tensorflow.python.ops import init_ops
+from tensorflow.python.ops import math_ops
+from tensorflow.python.ops import variable_scope
+from tensorflow.python.ops import script_ops
+from tensorflow.python.ops import variables
+from tensorflow.python.platform import test
+from tensorflow.python.training import device_setter
+from tensorflow.python.training import server_lib
+from tensorflow.python.util import compat
+from tensorflow.python.training import saver
+from tensorflow.python.ops import resources
+
+import tensorflow as tf
+
+
+class WarmStartHook(tf.estimator.SessionRunHook):
+
+  def __init__(self, ckpt_to_initialize_from, vars_to_warm_start):
+    self._ckpt_to_initialize_from = ckpt_to_initialize_from
+    self._vars_to_warm_start = vars_to_warm_start
+
+  def begin(self):
+    self._restore_op = de.warm_start(
+        ckpt_to_initialize_from=self._ckpt_to_initialize_from,
+        vars_to_warm_start=self._vars_to_warm_start)
+
+  def after_create_session(self, session, coord):
+    session.run(self._restore_op)
+
+
+@test_util.deprecated_graph_mode_only
+class WarmStartUtilTest(test.TestCase):
+
+  def _test_warm_start(self, num_shards, use_regex):
+    devices = ["/cpu:0" for _ in range(num_shards)]
+    ckpt_prefix = os.path.join(self.get_temp_dir(), "ckpt")
+    id_list = [x for x in range(100)]
+    val_list = [[x] for x in range(100)]
+
+    with self.session(graph=ops.Graph()) as sess:
+      embeddings = de.get_variable("embeddings",
+                                   dtypes.int64,
+                                   dtypes.float32,
+                                   devices=devices,
+                                   initializer=0.0)
+      ids = constant_op.constant(id_list, dtype=dtypes.int64)
+      vals = constant_op.constant(val_list, dtype=dtypes.float32)
+      self.evaluate(embeddings.upsert(ids, vals))
+      save = saver.Saver(var_list=[embeddings])
+      save.save(sess, ckpt_prefix)
+
+    with self.session(graph=ops.Graph()) as sess:
+      embeddings = de.get_variable("embeddings",
```

Comment:
recommend using different name for testing different name function in one case, or you'd better to write another case.